### DISCUSSION TITLE
With the latest version, the calling method will return 404

### DISCUSSION THREAD

**Discussion by q0015300153** [livewire/livewire#5462] [1. Help] [answered]
After I upgraded Laravel today,
```bash
composer update
```

It is found that the methods of livewire become uncallable,
`laravel/framework:9.45.1`
`livewire/livewire:2.10.8`
![image](https://user-images.githubusercontent.com/15045652/209294153-24e63ca2-8dc2-43f0-902f-c29613c707bd.png)

After rolling back to the old version, it can be used normally.
`laravel/framework:9.38.0`
`livewire/livewire:2.10.7`

my controller
`app\Http\Livewire\HelloWorld.php`
```php
namespace App\Http\Livewire;

use Livewire\Component;

class HelloWorld extends Component
{
    public string $name;

    public function render()
    {
        $name = "TB";
        return view('livewire.hello-world', compact(['name']));
    }
}
```

my component 
`resources\views\livewire\hello-world.blade.php`
```html
<div>
    My name is
    <input class="form-control" type="text" wire:model="name"><br>
    Hi~ My name is {{ $name }}
</div>
```

The result that looks like on the web page
![image](https://user-images.githubusercontent.com/15045652/209296743-a6feee58-77ed-4d3d-a084-519b7dd92d16.png)

Then when I change the name he gives 404 error
![image](https://user-images.githubusercontent.com/15045652/209295256-0502249f-5783-43c4-a7af-664a0700507d.png)
![image](https://user-images.githubusercontent.com/15045652/209295288-38e2f016-5c99-4dd3-8c4b-0da9cf480062.png)

After testing, it is confirmed that the error caused by livewire setting asset_url and app_url
`config\livewire.php`
```php
'asset_url' => '/test',
'app_url' => '/test',
```
But my system architecture needs to set these two parameters, and the old version can be executed normally
Because my Apache site settings
![image](https://user-images.githubusercontent.com/15045652/209297141-3203715e-4a75-4976-b480-aa1d42d7cfcf.png)

Executes normally without this setting
```php
'asset_url' => null,
'app_url' => null,
```
![image](https://user-images.githubusercontent.com/15045652/209296840-c9401b13-7a80-4b65-aa62-af08d8094244.png)
![image](https://user-images.githubusercontent.com/15045652/209297322-6b7be9df-44fa-4af8-9dbd-f01cc8c41a7d.png)
![image](https://user-images.githubusercontent.com/15045652/209297363-ba25a665-b9ba-4cb0-8979-51947768bb9a.png)
![image](https://user-images.githubusercontent.com/15045652/209297287-b3f0f650-2289-46ea-8645-8d3ce18c4a8f.png)

Does anyone know how to fix this issue please?

**Comment by joshhanley** [comment, marked as answer]
@q0015300153 there was a bug released in v2.10.8. I have submitted a PR #5490 with a fix. Hope this helps!

**Reply by q0015300153** [reply]
Thanks a lot, I'll try it out

**Reply by joshhanley** [reply]
@q0015300153 no worries! Let me know if there are any issues with it 🙂

**Reply by q0015300153** [reply]
@joshhanley Sorry, I still have a little problem after trying all day,
finally, I used a very ugly method to temporarily solve it (QAQ),
In your modified file.
`\src\Controllers\HttpConnectionHandler.php`
```php
$url     = Livewire::originalUrl();
$url     = Str::replaceFirst('/' . request('fingerprint')['locale'], '', $url);
$url     = Str::replaceFirst(request()->getBaseUrl(), '', $url);
$request = $this->makeRequestFromUrlAndMethod($url, Livewire::originalMethod());
````
![image](https://user-images.githubusercontent.com/15045652/211284157-b6c3fe27-4b83-4280-a25a-8f5174db6be7.png)

Then the system can operate normally,
if you need a case, I think it can be organized into a Dockerfile for you~

**Reply by joshhanley** [reply]
@q0015300153 thanks for testing. Can you comment on the PR (instead of here) with a break down of what still wasn’t working and a summary of what you changed and make it clear the difference (as I’m not sure from the above), that way it’s all contained in the same place? Thanks
